CMakeLists.txt

```diff
@@ -745,11 +750,31 @@ if(HPX_WITH_AGAS_DUMP_REFCNT_ENTRIES)
   hpx_add_config_define(HPX_HAVE_AGAS_DUMP_REFCNT_ENTRIES)
 endif()
 
+hpx_option(HPX_WITH_DISTRIBUTED_RUNTIME BOOL
+  "Enable the distributed runtime (default: ON). Turning off the distributed runtime completely disallows the creation and use of components and actions. Turning this option off is experimental!"
+  ON CATEGORY "Build Targets" ADVANCED)
+
+if(HPX_WITH_DISTRIBUTED_RUNTIME)
+  hpx_add_config_define(HPX_HAVE_DISTRIBUTED_RUNTIME)
+endif()
+
 # Should networking be supported?
 hpx_option(HPX_WITH_NETWORKING BOOL
   "Enable support for networking and multi-node runs (default: ON)"
   ON CATEGORY "Parcelport")
 
+if(HPX_WITH_NETWORKING AND NOT HPX_WITH_DISTRIBUTED_RUNTIME)
+  hpx_error("HPX_WITH_NETWORKING=ON requires HPX_WITH_DISTRIBUTED_RUNTIME=ON")
+endif()
+
+if(HPX_WITH_LOGGING AND NOT HPX_WITH_DISTRIBUTED_RUNTIME)
+  hpx_error("HPX_WITH_DISTRIBUTED_RUNTIME=OFF requires HPX_WITH_LOGGING=OFF")
+endif()
+
+if(HPX_WITH_APEX AND NOT HPX_WITH_DISTRIBUTED_RUNTIME)
+  hpx_error("HPX_WITH_DISTRIBUTED_RUNTIME=OFF requires HPX_WITH_APEX=OFF")
+endif()
```

Comment:
Do we really have to disable APEX here? I think disabling the performance counter related API in the APEX/HPX interface should be sufficient.

Reply:
APEX does depend on actions: https://github.com/khuck/xpress-apex/blob/6bc59aeb18ae5badce1c42fe35c9a15dcc382b06/src/apex/profiler_listener.cpp#L1728-L1756. That might be one of the only places though and making it work without the distributed runtime is probably doable.

Reply:
That doesn't seem to actually need an action as it launches things on the current locality only. I think this could be changed easily - @khuck?

Reply:
I'm confused...why does distributed require APEX to be disabled?  What's the context here?  Are actions deprecated?  What's the change required to just schedule the task locally?

Reply:
Kevin, actions are not deprecated. We're trying to enable a minimal configuration of HPX that essentially is reduced to purely local operation. In this case, actions and components would be not available. The code in APEX does not actually require actions as everything is limited to local execution. Instead of using async/apply to run the function that is associated with the action you could simply use async/apply to execute the function itself:
```
void apex_schedule_process_profiles() {
    if(get_hpx_runtime_ptr() == nullptr) return;
    if(!thread_instance::is_worker()) return;
    if(hpx_shutdown) {
        APEX_TOP_LEVEL_PACKAGE::profiler_listener::process_profiles_wrapper();
    } else {
        if(!consumer_task_running.test_and_set(memory_order_acq_rel)) {
            try {
                hpx::apply(&apex_internal_process_profiles);
            } catch(...) {
                // During shutdown, we can't schedule a new task,
                // so we process profiles ourselves.
                profiler_listener::process_profiles_wrapper();
            }
        }
    }
}
```

Reply:
And just to be explicit, the corresponding header: https://github.com/khuck/xpress-apex/blob/6bc59aeb18ae5badce1c42fe35c9a15dcc382b06/src/apex/profiler_listener.cpp#L70, would need to be `#include <hpx/local_async.hpp>` after that. The performance counters in APEX would at the moment have to be fully disabled though.

Reply:
I think this also doesn't have to resolved before this is merged.